### PR TITLE
Changes to SigmaJs and Dagre

### PR DESCRIPTION
**Pull request by davidlipo**

### CODE REVIEW

**Comment by vvakame** [comment]
*dagre/dagre.d.ts*

to author(@qinfchen). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

---

*sigmajs/sigmajs.d.ts*

to author(@qinfchen). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

File: sigmajs/sigmajs.d.ts [hunk 1]
```diff
@@ -15,10 +15,17 @@ declare module SigmaJs{
...
         readPrefix: string;
+        settings(setting: string) : any;
         x: number;
         y: number;
     }
 
+    interface Canvas {
+        edges: {[renderType: string]: Function};
+        labels: {[renderType: string]: Function};
+        nodes: {[renderType: string]: Function};
```
Comment (adidahiya) [inline sigmajs/sigmajs.d.ts]: Is it possible to be more specific about the `Function` types here? `Function` is a cop-out, similar to `any`.

File: sigmajs/sigmajs.d.ts [hunk 2]
```diff
@@ -39,6 +46,7 @@ declare module SigmaJs{
     }
 
     interface Edge {
+        [key : string] : any;
```
Comment (qinfchen) [inline sigmajs/sigmajs.d.ts]: I am not sure about this

File: sigmajs/sigmajs.d.ts [hunk 3]
```diff
@@ -149,7 +163,7 @@ declare module SigmaJs{
 
     interface SigmaConfigs {
         container?: Element;
-        graph?: Graph;
+        graph?: GraphData;
```
Comment (qinfchen) [inline sigmajs/sigmajs.d.ts]: why this change? graph object has many properties, this will break all the graph related stuff

File: sigmajs/sigmajs.d.ts (hunk 1 above)
Comment (davidlipo) [inline sigmajs/sigmajs.d.ts]: Done.

File: sigmajs/sigmajs.d.ts [hunk 4]
```diff
@@ -269,8 +285,19 @@ declare module SigmaJs{
...
+    interface SVGObject<T> {
+        create: (object: T, ...a:any[]) => Element;
+        update: (object: T, ...a:any[]) => void;
+    }
 }
 
 declare var sigma: SigmaJs.SigmaFactory;
 declare var CustomShapes: SigmaJs.CustomShapes;
-declare var ShapeLibrary: SigmaJs.CustomShapes;
+declare var ShapeLibrary: SigmaJs.CustomShapes;
```
Comment (qinfchen) [inline sigmajs/sigmajs.d.ts]: new line

File: dagre/dagre.d.ts [hunk 5]
```diff
@@ -10,20 +10,24 @@ declare module Dagre{
...
+    }
+
     interface GraphLib {
         Graph: Graph;
     }
 }
 
-declare var dagre: Dagre.DagreFactory;
-
+declare var dagre: Dagre.DagreFactory;
```
Comment (qinfchen) [inline dagre/dagre.d.ts]: new line

File: dagre/dagre.d.ts [hunk 6]
```diff
@@ -10,20 +10,24 @@ declare module Dagre{
 
     interface Graph {
         new (): Graph;
-        edges(): string[];
-        edge(id: string): any;
+        edges(): Edge[];
+        edge(id: any): any;
```
Comment (qinfchen) [inline dagre/dagre.d.ts]: why any?

File: sigmajs/sigmajs.d.ts (hunk 4 above)
Comment (davidlipo) [inline sigmajs/sigmajs.d.ts]: Done.

File: dagre/dagre.d.ts (hunk 5 above)
Comment (davidlipo) [inline dagre/dagre.d.ts]: Done.

File: dagre/dagre.d.ts (hunk 6 above)
Comment (davidlipo) [inline dagre/dagre.d.ts]: Because it turns out that you can also input an object in to edge() to get an edge returned. In certain cases, that's the only way to access some edge details.

**Comment by davidlipo** [comment]
Bump?

**Comment by qinfchen** [comment]
looks good to me.

**Comment by vvakame** [comment]
@qinfchen thanks mate!
@adidahiya sorry, It merges. please pull request if you encounter trouble.
@davidlipo thanks mate!
